Add membership and commit rights document

mrocklin opened the pull request:
This was previously discussed in https://github.com/dask/dask/issues/3223

mrocklin committed:
Add membership and commit rights document

stsievert commented on membership.md:
```diff
@@ -0,0 +1,90 @@
+
+1.  The owner thinks it's a good idea.
+2.  Some other owner employed by a different institution thinks it's a good
+    idea.
```
Maybe

``` diff
- 1.  The owner thinks it's a good idea.
- 2.  Some other owner employed by a different institution thinks it's a good
-    idea.
+ 1. At least two owners from separate employers agree ownership is a good idea.
```

mrocklin replied:
Done

stsievert commented on membership.md:
```diff
@@ -0,0 +1,90 @@
+------
+
+There is a set of individuals who have ownership privileges over the entire
+GitHub organization.  They have commit rights to all repositories by default
```
> They have commit rights to all repositories by default

By GitHub default? By Dask org default?

mrocklin replied:
This is GitHub's policy

stsievert commented on membership.md:
```diff
@@ -0,0 +1,90 @@
+We acknowledge that write permissions to individual projects may be better handled by teams.  Team membership will be handled similar to write permissions.
+
+
+Removing permissions
```
> Removing permissions

Removing write permissions?

mrocklin replied:
I ended up not changing this.  It's not just write permissions, it's lots of things, like the ability to add new members, repositories, and so on.

guillaumeeb commented on membership.md:
```diff
@@ -0,0 +1,90 @@
+4.  The new member demonstrates some degree of familiarity of open source
+    practices and online courtesy.
+
+Members can be given write privileges to any repository by an owner by the
```
Does this mean that a contributor has to be a Dask org member before giving him/her write priviledges?

mrocklin replied:
I did not intend to imply that.  We do have cases today where people have write priveleges without being members of the org.  Often this happens when we bring in an external repository.  

I'll change the language in a bit.

guillaumeeb reviewed (commented):
Just asking some clarification on the necessity to become a member first before being given write priviledges.

Also, there are some dots missing at the end of sentences in __Removing permissions__

mrocklin committed:
edits from review

mrocklin commented:
Thanks @guillaumeeb and @stsievert for the review

mrocklin commented:
If there are no further comments then I suggest that we merge this in 48 hours.  All current owners have been informed of this PR and the intent to merge.

martindurant commented:
+1 to merging

jakirkham commented on membership.md:
```diff
@@ -0,0 +1,91 @@
+Individuals can be given write privileges to any repository by an owner by the
+above conditions and the following:
+
+1.  The lead maintainer of that repository thinks it's a good idea
```
How do we decide who is the lead maintainer?

mrocklin replied:
I think that in practice we generally just know.  Happy to add something here if you have a suggestion.

jakirkham replied:
I think it was "lead" that was confusing to me. For instance I don't know the lead maintainer of dask-jobqueue, but know a few people that are heavily involved. Have pushed some text to relax this to maintainers instead of lead. Though am happy to discuss further as needed.

jakirkham commented on membership.md:
```diff
@@ -0,0 +1,91 @@
+permissions
+
+Any owner may temporarily remove permissions of any non-owner if there is some
+pressing reason to do so, such as insecure credentials.  A consensus of owners
```
Am trying to better understand how "consensus" is used here. Does it mean lazy consensus, unanimous, majority with some percentage, something else?

mrocklin replied:
As elsewhere, we've been using lazy consensus.  Is there a change that you would like us to make here?

jakirkham replied:
I think we are already having a [discussion in this thread]( https://github.com/dask/governance/pull/11#pullrequestreview-227868480 ). We can update here and elsewhere once we have reached a conclusion.

jakirkham replied:
I've pushed a similar change as was used there (e.g. copying from the membership addition policy above). Please let me know your thoughts.

jakirkham commented on membership.md:
```diff
@@ -0,0 +1,91 @@
+
+Any owner may temporarily remove permissions of any non-owner if there is some
+pressing reason to do so, such as insecure credentials.  A consensus of owners
+can remove permissions of anyone long-term
```
```suggestion
can remove permissions of anyone long-term.
```

jakirkham reviewed (commented):
Thanks for working on this Matt. Generally this seems good. Had a couple questions.

jakirkham committed:
Update membership.md

Co-Authored-By: mrocklin <mrocklin@gmail.com>

jakirkham committed:
Apply same criteria to removals as additions

jakirkham committed:
Relax lead maintainer for adding individuals

Ideally everyone that is fairly actively involved in a project should be
on board with a new addition.

jakirkham commented:
Have pushed a few changes to address things raised here. Please let me know your thoughts on these changes.

TomAugspurger reviewed (approved):
Let's do another 48-hour call for +/-1 on this. If there aren't any additional concerns then we can merge.

martindurant commented:
+1 on all of this from me

mmccarty commented:
+1 here

mrocklin commented:
Merging in 24 hours if there are no further comments

jakirkham commented:
Thanks all! 😄
